tempest/api/workloadmgr/workload/test_workload_reset.py

```diff
@@ -0,0 +1,112 @@
+from tempest import command_argument_string
+from tempest.util import cli_parser
+from tempest.util import query_data
+
+LOG = logging.getLogger(__name__)
+CONF = config.CONF
+
+class WorkloadTest(base.BaseWorkloadmgrTest):
+
+    credentials = ['primary']
+    volumes = None
+
+    @classmethod
+    def setup_clients(cls):
+        super(WorkloadTest, cls).setup_clients()
+        cls.client = cls.os.wlm_client
+        reporting.add_test_script(str(__name__))
+    
+    def test_workload_reset(self):
+        try:
+
+            ## VM and Workload ###
+
+            mount_points = ["mount_data_a", "mount_data_b"] 
+
+            vm_id = self.create_vm(vm_cleanup=True)
+            LOG.debug("VM ID : "+str(vm_id))
+
+            volume_id = self.create_volume(volume_cleanup=True)
+            LOG.debug("Volume ID: " + str(volume_id))
+            volumes = tvaultconf.volumes_parts
+    
+            self.attach_volume(volume_id, vm_id, attach_cleanup=True)
+            LOG.debug("Volume attached")
+        
+            LOG.debug("Sleeping for 40 sec")
+            time.sleep(40)
+            i=1
+            workload_id = self.workload_create([vm_id],tvaultconf.parallel, workload_name="w1", workload_cleanup=True, description='test')
+            LOG.debug("Workload ID: " + str(workload_id))
+            if(workload_id != None):
+                self.wait_for_workload_tobe_available(workload_id)
+                if(self.getWorkloadStatus(workload_id) == "available"):
+                    reporting.add_test_step("Create workload-{}".format(i), tvaultconf.PASS)
+                else:
+                    reporting.add_test_step("Create workload-{}".format(i), tvaultconf.FAIL)
+                    reporting.set_test_script_status(tvaultconf.FAIL)
+            else:
+                reporting.add_test_step("Create workload-{}".format(i), tvaultconf.FAIL)
+                reporting.set_test_script_status(tvaultconf.FAIL)
+                raise Exception("Workload creation failed")
+
+            ### Full snapshot ###
+
+            self.created = False
+
+            snapshot_id = self.workload_snapshot(workload_id, True, snapshot_name="Snap1", snapshot_cleanup=True)
+
+            time.sleep(5)
+            self.wait_for_workload_tobe_available(workload_id)
+            if(self.getSnapshotStatus(workload_id, snapshot_id) == "available"):
+                reporting.add_test_step("Create full snapshot-{}".format(i), tvaultconf.PASS)
+                LOG.debug("Full snapshot available!!")
+            else:
+                reporting.add_test_step("Create full snapshot-{}".format(i), tvaultconf.FAIL)
+                raise Exception("Snapshot creation failed")
+
+            i+=1
+            volsnaps_before = self.get_volume_snapshots(volume_id)
+
+            LOG.debug("\nVolume snapshots after full snapshot : {}\n".format(volsnaps_before))
+           
+            self.workload_reset(workload_id)
+ 
+            
+            self.created = False
```

Comment:
Is this variable used ? If not, remove it.

Reply:
done